data-document-model/src/main/java/io/micronaut/data/document/model/query/builder/MongoQueryBuilder.java

```diff
@@ -218,6 +218,19 @@ public final class MongoQueryBuilder implements QueryBuilder {
         addCriterionHandler(QueryModel.Contains.class, (context, obj, criterion) -> {
             handleRegexPropertyExpression(context, obj, criterion, criterion.isIgnoreCase(), false, false, false);
         });
+        addCriterionHandler(QueryModel.ArrayContains.class, (context, obj, criterion) -> {
```

Comment:
`ArrayContains` might be a bit confusing for people not using an array as the actual type but a collection. Maybe we can just have `contains` and decide based on the type/association?

Reply:
We already have Contains 
```
addCriterionHandler(QueryModel.Contains.class, (context, obj, criterion) -> {
            handleRegexPropertyExpression(context, obj, criterion, criterion.isIgnoreCase(), false, false, false);
        });
```
ant that is used for Strings everywhere. On the other hand we have already ArrayContains for Azure Cosmos Db so it makes sense to call it the same here I think, we can explain in the docs and examples how to use it with list or array.

Reply:
Ok so maybe we can also have the alias `collectionContains`

Reply:
Added, hoping it won't cause more confusion:/